torch/serialization.py

```diff
@@ -568,17 +601,21 @@ def persistent_load(saved_id):
         raise RuntimeError("Invalid protocol version: %s" % protocol_version)
 
     _sys_info = pickle_module.load(f, **pickle_load_args)
-    unpickler = pickle_module.Unpickler(f, **pickle_load_args)
-    unpickler.persistent_load = persistent_load
-    result = unpickler.load()
+
+    # Read through the data pickle without actually loading anything, just
+    # continue through it until a STOP opcode
+    data_start = f.tell()
+    [_ for _ in pickletools.genops(f)]
```

Comment:
so the problem is that we rely here that tensor storages will be laid out in the same order as in pickle archive, right?

can't we just iterate "smarter" through pickle's format, find all the "storage" records and retrieve data type and size from them? that should be pretty robust.